### PR TITLE
vm/plugins

### PR DESCRIPTION
**Pull request by parvdesai-tenable**
Contains changes related to search method for vm/plugins

### CODE REVIEW

File: tenable/io/v3/vm/plugins/api.py [hunk 1]
```diff
@@ -10,39 +10,110 @@
...
+                    ...  }],
+                    ...  'and': [{
+                    ...      'value': '3',
+                    ...      'operator': 'oper',
+                    ...      'property': 3
+                    ...  }]
+                    ... }
+                As the filters may change and sortable fields may change over
+                time, it's highly recommended that you look at the output of
+                the :py:meth:`tio.v3.vm.filters.families_filters()`
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/plugins/api.py]:
```suggestion
                the :py:meth:`tio.v3.definitions.vm.plugin_families()`
```

File: tenable/io/v3/vm/plugins/api.py [hunk 2]
```diff
@@ -10,39 +10,110 @@
...
+                               resource='families',
+                               **kwargs
+                               )
 
     def family_details(self, family_id: int) -> Dict:
-        """
+        '''
         Retrieve the details for a specific plugin family.
 
         :devportal:`plugins: family-details plugins-family-details>`
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/plugins/api.py]: Typing is missing from `family_id` parameter's description in docstring. Return type should be `Dict` in docstring.

File: tenable/io/v3/vm/plugins/api.py [hunk 3]
```diff
@@ -10,39 +10,110 @@
...
+    _path = 'api/v3/plugins'
     _conv_json = True
 
-    def families(self) -> List:
-        """
-        List the available plugin families.
-
-        :devportal:`plugins: families <plugins-families>`
-
+    def search_families(self, **kwargs) -> Union[CSVChunkIterator,
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/plugins/api.py]: Can we alphabetize the methods?

File: tenable/io/v3/vm/plugins/api.py [hunk 4]
```diff
@@ -58,11 +129,11 @@ def family_details(self, family_id: int) -> Dict:
...
-        return self._get(f"families/{family_id}")
+        '''
+        return super()._details(f'families/{family_id}')
 
     def plugin_details(self, plugin_id: int) -> Dict:
-        """
+        '''
         Retrieve the details for a specific plugin.
 
         :devportal:`plugins: plugin-details <plugins-plugin-details>`
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/plugins/api.py]: Return type should be Dict in docstring.

File: tenable/io/v3/vm/plugins/api.py [hunk 5]
```diff
@@ -78,10 +149,90 @@ def plugin_details(self, plugin_id: int) -> Dict:
...
+                    ...  }],
+                    ...  'and': [{
+                    ...      'value': '3',
+                    ...      'operator': 'oper',
+                    ...      'property': 3
+                    ...  }]
+                    ... }
+                As the filters may change and sortable fields may change over
+                time, it's highly recommended that you look at the output of
+                the :py:meth:`tio.v3.vm.filters.plugins_filters()`
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/plugins/api.py]:
```suggestion
                the :py:meth:`tio.v3.definitions.vm.plugins()`
```

File: tests/io/v3/vm/plugins/objects.py [hunk 6]
```diff
@@ -0,0 +1,249 @@
+PLUGIN_ID = 22372
+PLUGIN = {
+    "attributes": [
```
Comment (harshshah-tenable) [inline tests/io/v3/vm/plugins/objects.py]: Please update `"` -> `'` at all instances

**Review by ngaglani-tenable — APPROVED** [review]

File: tenable/io/v3/vm/plugins/api.py [hunk 7]
```diff
@@ -78,10 +64,175 @@ def plugin_details(self, plugin_id: int) -> Dict:
...
+                Examples:
+                    >>> [('field_name_1', 'asc'),
+                    ...      ('field_name_2', 'desc')]
+            limit (int, optional):
+                Number of objects to be returned in each request.
+                Default and max_limit is 200.
+            next (str, optional):
+                The pagination token to use when requesting the next page of
+                results. This token is presented in the previous response.
+            return_resp (bool, optional):
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/plugins/api.py:127]: Update docstring.

File: tenable/io/v3/vm/plugins/api.py [hunk 8]
```diff
@@ -78,10 +64,175 @@ def plugin_details(self, plugin_id: int) -> Dict:
...
+                specific page of data.
+            return_csv (bool, optional):
+                If set to true, it will return the CSV response or
+                iterable (based on return_resp flag). Iterator returns all
+                rows in text/csv format for each call with row headers.
+        Returns:
+            iterable:
+                The iterable that handles the pagination for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/plugins/api.py]: Update docstring.

File: tenable/io/v3/vm/plugins/api.py [hunk 9]
```diff
@@ -78,10 +64,175 @@ def plugin_details(self, plugin_id: int) -> Dict:
...
+                Examples:
+                    >>> [('field_name_1', 'asc'),
+                    ...      ('field_name_2', 'desc')]
+            limit (int, optional):
+                Number of objects to be returned in each request.
+                Default and max_limit is 200.
+            next (str, optional):
+                The pagination token to use when requesting the next page of
+                results. This token is presented in the previous response.
+            return_resp (bool, optional):
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/plugins/api.py:211]: Update docstring.

File: tenable/io/v3/vm/plugins/api.py [hunk 10]
```diff
@@ -78,10 +64,175 @@ def plugin_details(self, plugin_id: int) -> Dict:
...
+                specific page of data.
+            return_csv (bool, optional):
+                If set to true, it will return the CSV response or
+                iterable (based on return_resp flag). Iterator returns all
+                rows in text/csv format for each call with row headers.
+        Returns:
+            iterable:
+                The iterable that handles the pagination for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/plugins/api.py]: update docstring.

**Review by Mangusingh-te — CHANGES_REQUESTED** [review]

File: tenable/io/v3/vm/plugins/api.py (hunk 8 above)
Comment (alpesh-te) [reply tenable/io/v3/vm/plugins/api.py]: Addressed

File: tenable/io/v3/vm/plugins/api.py (hunk 9 above)
Comment (alpesh-te) [reply tenable/io/v3/vm/plugins/api.py:211]: Addressed

File: tenable/io/v3/vm/plugins/api.py (hunk 10 above)
Comment (alpesh-te) [reply tenable/io/v3/vm/plugins/api.py]: Addressed

**Review by alpesh-te — APPROVED** [review]
LGTM!

**Pull request merged**